fractal_tasks_core/tasks/apply_registration_to_ROI_tables.py

```diff
@@ -0,0 +1,312 @@
+        # Update list of available tables & its metadata
+        current_tables = group_tables.attrs.asdict().get("tables") or []
+        new_tables = current_tables + [new_roi_table]
+        group_tables.attrs["tables"] = new_tables
+        group_tables[roi_table].attrs["type"] = "ngff:region_table"
+
+    # TODO: Optionally apply registration to other tables as well?
+    # e.g. to well_ROI_table based on FOV_ROI_table
+    # => out of scope for the initial task, apply registration separately
+    # to each table
+    # Easiest implementation: Apply average shift calculcated here to other
+    # ROIs. From many to 1 (e.g. FOV => well) => average shift, but crop len
+    # From well to many (e.g. well to FOVs) => average shift, crop len by that
+    # amount
+    # Many to many (FOVs to organoids) => tricky because of matching
+
+    return {}
+
+
+# Helper functions
+def get_acquisition_paths(zattrs: dict) -> dict[int, str]:
+    """
+    Create mapping from acquisition indices to corresponding paths.
+
+    FIXME: this looks like a function that should be part of
+    ../lib_read_fractal_metadata.py.
+
+    Attrs:
+        zattrs:
+            Attributes of a plate zarr group.
+
+    Returns:
+        Dictionary with `(acquisition index: image path)` key/value pairs.
+    """
+    acquisition_dict = {}
+    for image in zattrs["well"]["images"]:
+        if "acquisition" not in image:
+            raise ValueError(
+                "Cannot get acquisition paths for Zarr files without "
+                "'acquisition' metadata at the well level"
+            )
+        if image["acquisition"] in acquisition_dict:
+            raise NotImplementedError(
+                "This task is not implemented for wells with multiple images "
+                "of the same acquisition"
+            )
+        acquisition_dict[image["acquisition"]] = image["path"]
+    return acquisition_dict
+
+
+def add_zero_translation_columns(ad_table: ad.AnnData):
+    """
+    Add three zero-filled columns (`translation_{x,y,z}`) to an AnnData table.
+    """
+    columns = ["translation_z", "translation_y", "translation_x"]
+    if ad_table.var.index.isin(columns).any().any():
+        raise ValueError(
+            "The roi table already contains translation columns. Did you "
+            "enter a wrong reference cycle?"
+        )
+    df = pd.DataFrame(np.zeros([len(ad_table), 3]), columns=columns)
+    df.index = ad_table.obs.index
+    ad_new = ad.concat([ad_table, ad.AnnData(df)], axis=1)
+    return ad_new
+
+
+def calculate_min_max_across_dfs(tables_list):
+    # Initialize dataframes to store the max and min values
+    max_df = pd.DataFrame(
+        index=tables_list[0].index, columns=tables_list[0].columns
+    )
+    min_df = pd.DataFrame(
+        index=tables_list[0].index, columns=tables_list[0].columns
+    )
+
+    # Loop through the tables and calculate max and min values
+    for table in tables_list:
+        if max_df is None:
```

Comment:
I do not understand this `if max_df is None` check. How could `max_df` be `None`?
First, we set it to a `DataFrame` at the beginning of the function. Then, within the `for` loop, we always set it to another `DataFrame`. It looks like this branch can never be executed.

I guess this is a leftover from a previous implementation. I think we should either remove the `if/else` check, or replace the initialization with `max_df=None`.

Reply:
Agreed, must have been a refactor leftover. I removed it and things work as expected, see https://github.com/fractal-analytics-platform/fractal-tasks-core/pull/487/commits/6d15f4f6a7e295ce070eab112abb7ede5bcdb54a